### ISSUE TITLE
Unable to remove device after.

### ISSUE THREAD

**Issue by ThorodanBrom** [linuxmint/blueberry#35]
I had paired my OnePlus X phone a while back, and the application worked like a charm. After that instance however, it has never worked. On clicking 'Connection', the button moves back to OFF every time. 
On clicking, 'Remove Device', it gives a confirmation, after clicking yes, the device is still there.

The first time I was able to pair my phone. After it stopped working, I unpaired my PC from the phone, and am unable to initiate a pair from the phone anymore. I think this is due to Blueberry still detecting that I'm paired or something like that.

![1](https://cloud.githubusercontent.com/assets/9331293/26190669/4323c9a6-3bc8-11e7-87d1-fcdf0a6d0dc7.png)
![2](https://cloud.githubusercontent.com/assets/9331293/26190670/43271bf6-3bc8-11e7-9340-e7bf55e4474d.png)

**Comment by Rudertier** [comment]
Dear all, 
I am experiencing the same issue, with both my phone and a bluetooth speaker. I am running Linuxmint 18.2. Is there any way to make a hard reset on blueberry?
Cheers

**Comment by deutrino** [comment]
I'm also having this problem, there is a phone from ages ago which I can't delete to save my life.

**Comment by NikoKrause** [comment]
Hey, can anybody of you, who experience this try to turn on bluetooth and then use the following command in a terminal:
```
rfkill unblock bluetooth
```

Are you able to pair to or remove the device after that?

**Comment by ThorodanBrom** [comment]
Yes, this seems to work, but I have to do this every single time I want to connect to my phone. I had earlier tried using Blueman, but that didn't work. This is the best solution yet!

**Comment by NikoKrause** [comment]
Yes. I have this issue too.
Can you try this version:

[blueberry_sleep.zip](https://github.com/linuxmint/blueberry/files/1462327/blueberry_sleep.zip)

**Comment by NikoKrause** [comment]
Also, which bluetooth adapter do you have?

**Comment by ThorodanBrom** [comment]
Hello, NikoKrause.

I am not able to determine my adapter; I've tried using lsusb, lspci, and inxi, but the Bluetooth adapter is not explicitly mentioned.

If you don't mind me asking, how am I supposed to test the Blueberry .deb file that you sent, without interfering with the regular install? Can I use chroot, or should I use a VM? (I'm a relative novice at Linux, but I am more that happy to help)

**Comment by NikoKrause** [comment]
Just unzip and double click to install.

You can downgrade blueberry to your older version at any time. I'll take later screenshots on how to do this.

Is your Bluetooth built-in or external? I have this issue with a built-in adapter in my Dell Latitude.

**Comment by NikoKrause** [comment]
@ThorodanBrom 
Here's how you downgrade blueberry to the repository version:
`Open Menu -> Software Sources -> Maintenance -> Downgrade foreign packages -> (tick) blueberry -> (press) Downgrade`

![screenshot-area-2017-11-11-170333](https://user-images.githubusercontent.com/8415124/32691067-afd6afda-c702-11e7-8858-7b6e248c3438.png)

**Comment by ThorodanBrom** [comment]
Hello Niko, sorry for the delay. I have installed the version you sent, and here are my observations.

-> On turning Bluetooth ON in Blueberry, all devices that I have tried to connect to previously appear. Note that new devices that appear on my phone **do not** appear on Blueberry. I am also unable to connect to my phone. It appears as _Not Set Up_ in Blueberry, however, on the phone, my laptop is listed as a paired device. When I try to initiate the connection from my phone, it says 'Connecting' and then just stops after a while.

-> On executing `rfkill unblock bluetooth` while Blueberry is open, a new device called _All types_ appears, and then the new devices that were not appearing earlier appear now. I am able to connect my phone to the laptop successfully. (Note, I was even able to remove my phone from the list in Blueberry after this, using the _Remove Device_ option)

-> On turning Bluetooth OFF and ON in Blueberry, it reverts back to how it was in the first point, however, it says _Disconnected_ for my phone, rather than _Not Set Up_. When I click my phone's name on the listing, and try to turn Connection ON, it refuses to do so, and additionally, I am unable to remove the phone from the listing.

It seems that Blueberry only works after the command you suggested, even with the version you sent. It is listed as ver 1.1.18 in Software Sources. I had initially installed the software and tried the above out. I thought it didn't work because I didn't do a restart, but there were no changes after the restart also.

I have an inbuilt Bluetooth adapter in my Toshiba Satellite Pro.

**Comment by NikoKrause** [comment]
@ThorodanBrom 
Can you give me the following info.

1. Turn bluetooth on and type in the command:
```
rfkill list bluetooth
```

2. Turn bluetooth on and if bluetooth does not work type in the command
```
rfkill list bluetooth
```

3. When bluetooth is turned on and does not work, make it work with:
```
rfkill unblock bluetooth
```
and show the output of
```
rfkill list bluetooth
```

**Comment by ThorodanBrom** [comment]
@NikoKrause , as you requested

1. When BT is On, and `rfkill list bluetooth` 

![image](https://user-images.githubusercontent.com/9331293/32852626-730c2126-ca30-11e7-9942-49fbc4b443c6.png)

2. It didn't work, so the same output as the previous applies.

3. Made it work with `rfkill unblock bluetooth`, and then `rfkill list bluetooth` 

![image](https://user-images.githubusercontent.com/9331293/32852726-be65539a-ca30-11e7-943c-9c7dace8489c.png)

4. And interestingly, when BT is Off, and `rfkill list bluetooth`

![image](https://user-images.githubusercontent.com/9331293/32852873-4911e8fa-ca31-11e7-9bf3-4f4dde40836f.png)

I've no idea why 2 modules are showing up when BT is On, I'm pretty sure that I have only 1 built-in module!

Note that turning on BT in all cases was done in Blueberry itself, not via command line. I have a keyboard shortcut, but it doesn't seem to work in Linux. It seems that only 'Toshiba Bluetooth' is unblocked in Blueberry, not the 'hci0' module. That gets unblocked only via the unblock command

**Comment by NikoKrause** [comment]
OK. I have the same as you. Two, when turned on and one when turned off.

**Comment by ThorodanBrom** [comment]
@NikoKrause, I'm not sure how helpful this is, but I found an issue raised in TLP that mentions it might be a kernel-related problem. https://github.com/linrunner/TLP/issues/180

**Comment by lloricode** [comment]
same issue as mine

**Comment by JamesTheAwesomeDude** [comment]
I'm also getting this. The underlying error seems to be `org.bluez.Error.NotReady`, but **blueberry isn't reporting this**.

```
[dead@CNU352DG8J ~]$ rfkill unblock bluetooth
[dead@CNU352DG8J ~]$ rfkill list
0: phy0: Wireless LAN
	Soft blocked: no
	Hard blocked: no
3: hci0: Bluetooth
	Soft blocked: no
	Hard blocked: no
[dead@CNU352DG8J ~]$ bluetoothctl power on
Changing power on succeeded
[dead@CNU352DG8J ~]$ bluetoothctl remove FC:D4:36:B3:6D:EE
Failed to remove device: org.bluez.Error.NotReady
```

I think that displaying the underlying errors when an action the user takes fails would be extremely useful
